x/incentives/keeper/distribute_test.go

```diff
@@ -1621,3 +1552,430 @@ func (s *KeeperTestSuite) TestSyncGroupGaugeWeights() {
+		},
+
+		"5: no coins to distribute (the FilledEpoch is still updated)": {
+			groups: []groupConfig{
+				{
+					group:      defaultGroupGauge,
+					groupGauge: withCoinsToDistribute(deepCopyGauge(nonPerpetualGauge), defaultCoins, defaultCoins),
+				},
+			},
+
+			volumeToSet: balancerAndCLVolumeConfig,
+		},
+
+		//// skipping
+
+		// skipping on sync failure
+		"6: skipped: synching fails due to no volume set": {
+			groups: []groupConfig{
+				{
+					group:      defaultGroupGauge,
+					groupGauge: withCoinsToDistribute(deepCopyGauge(nonPerpetualGauge), defaultCoins, defaultCoins),
+
+					expectedSkipped: true,
+				},
+			},
+
+			volumeToSet: []osmomath.Int{},
+		},
+
+		// skipping on gauge being inactive
+		"7: skipping on gauge being inactive": {
+			groups: []groupConfig{
+				{
+					group: defaultGroupGauge,
+					// This makes non-perpetual gauge inactive
+					groupGauge: withNonPerpetualEpochs(deepCopyGauge(nonPerpetualGauge), 1, 1),
+
+					expectedSkipped: true,
+				},
+			},
+
+			volumeToSet: balancerAndCLVolumeConfig,
+		},
+
+		// skipping because this gauge has no pool associated with it.
+		// we only distributed to internal gauges.
+		"8: associated group gauge is non perpetual and finished": {
+			groups: []groupConfig{
+				{
+					group:      groupToInvalidUnderlying,
+					groupGauge: perpetualGauge,
+
+					expectedSkipped: true,
+				},
+			},
+
+			volumeToSet: balancerOnlyVolumeConfig,
+		},
+
+		///////////////// multi-gauges
+
+		// Note that groups distribute to overlapping gauges.
+		"9: multiple groups with varying number of underlying gauges": {
+			groups: []groupConfig{
+				{
+					group:      singleRecordGroupGauge,
+					groupGauge: perpetualGauge,
+				},
+				{
+					group:      withGroupGaugeId(defaultGroupGauge, 6),
+					groupGauge: withGaugeId(deepCopyGauge(nonPerpetualGauge), 6),
+				},
+			},
+
+			volumeToSet: []osmomath.Int{defaultGroupGauge.InternalGaugeInfo.GaugeRecords[0].CumulativeWeight, defaultGroupGauge.InternalGaugeInfo.GaugeRecords[1].CumulativeWeight},
+		},
+
+		"10: skipping one does not fail the other": {
+			groups: []groupConfig{
+				{
+					group: defaultGroupGauge,
+					// This makes non-perpetual gauge inactive
+					groupGauge: withNonPerpetualEpochs(deepCopyGauge(nonPerpetualGauge), 1, 1),
+
+					// skipped because inactive
+					expectedSkipped: true,
+				},
+				{
+					group:      withGroupGaugeId(defaultGroupGauge, 6),
+					groupGauge: withGaugeId(deepCopyGauge(nonPerpetualGauge), 6),
+				},
+			},
+
+			volumeToSet: balancerAndCLVolumeConfig,
+		},
+
+		///////////////// error cases
+
+		"11: not enough funds in the module account causing error in AddToGaugeRewards": {
+			groups: []groupConfig{
+				{
+					group:      singleRecordGroupGauge,
+					groupGauge: deepCopyGauge(perpetualGauge),
+				},
+			},
+
+			shouldSkipFunding: true,
+
+			volumeToSet: balancerOnlyVolumeConfig,
+		},
+
+		// TODO: even splitting policy test cases once supported.
```

Comment:
Note to reveiewer: tracker here https://github.com/osmosis-labs/osmosis/issues/6435

Reply:
I would be in support of removing even split policy altogether as it seems to have been added as a placeholder before volume splitting policy was implemented

Reply:
Agreed: https://github.com/osmosis-labs/osmosis/issues/6500